src/plugins/typechecker/README.md

```diff
@@ -0,0 +1,115 @@
+
+Currently there is no way to check the specification automatically upon mounting. Simply 
+retrieve a random key from the specification using `kdb get <path>` will cause
+the typechecking to happen, issuing a warning if it detects any problem with it.
+
+## Examples
+
+Create a sample configuration specification with three keys. As it is valid,
+there will be no error issued. Currently we include the type signatures of the
+various keywords manually in a specification. We use kdb shell to delay the 
+typechecking until we have finished writing the whole specification.
+
+```sh
+# Backup-and-Restore:spec/examples/simplespecification
+
+sudo kdb mount simplespecification.ini spec/examples/simplespecification ini typechecker
+
+echo 'kdbGet spec/examples/simplespecification \
+keySetName spec/examples/simplespecification/elektra/spec/fallback/# \
+keySetMeta elektra/spec/type RegexContains b a => Key b :: . -> Key a -> Key a \
+keySetMeta elektra/spec/impl fallback a (Key P.Nothing) = a \n fallback _ a = a \
+ksAppendKey \
+keyClear \
+keySetName spec/examples/simplespecification/elektra/spec/override/#\
+keySetMeta elektra/spec/type RegexContains b a => Key b :: . -> Key a -> Key a \
+keySetMeta elektra/spec/impl override (Key P.Nothing) b = b \n override a _ = a \
+ksAppendKey \
+keyClear \
+keySetName spec/examples/simplespecification/elektra/spec/check/range \
+keySetMeta elektra/spec/type RegexIntersects a b => P.Proxy b :: Range . -> Key a -> Key (RegexIntersection a b) \
+keySetMeta elektra/spec/impl checkrange _ a = a \
+ksAppendKey \
+keyClear \
+keySetName spec/examples/simplespecification/elektra/spec/check/long \
+keySetMeta elektra/spec/type RegexIntersects a "-[1-9]|-214748364[0-8]|-?[1-9][0-9]|-?[1-9][0-9][0-9]|-?[1-9][0-9][0-9][0-9]|-?[1-9][0-9][0-9][0-9][0-9]|-?[1-9][0-9][0-9][0-9][0-9][0-9]|-?[1-9][0-9][0-9][0-9][0-9][0-9][0-9]|-?[1-9][0-9][0-9][0-9][0-9][0-9][0-9][0-9]|-?[1-9][0-9][0-9][0-9][0-9][0-9][0-9][0-9][0-9]|-?1[0-9][0-9][0-9][0-9][0-9][0-9][0-9][0-9][0-9]|-?20[0-9][0-9][0-9][0-9][0-9][0-9][0-9][0-9]|-?21[0-3][0-9][0-9][0-9][0-9][0-9][0-9][0-9]|-?214[0-6][0-9][0-9][0-9][0-9][0-9][0-9]|-?2147[0-3][0-9][0-9][0-9][0-9][0-9]|-?21474[0-7][0-9][0-9][0-9][0-9]|-?214748[0-2][0-9][0-9][0-9]|-?2147483[0-5][0-9][0-9]|-?21474836[0-3][0-9]|[0-9]|214748364[0-7]" => Key a -> Key (RegexIntersection a "-[1-9]|-214748364[0-8]|-?[1-9][0-9]|-?[1-9][0-9][0-9]|-?[1-9][0-9][0-9][0-9]|-?[1-9][0-9][0-9][0-9][0-9]|-?[1-9][0-9][0-9][0-9][0-9][0-9]|-?[1-9][0-9][0-9][0-9][0-9][0-9][0-9]|-?[1-9][0-9][0-9][0-9][0-9][0-9][0-9][0-9]|-?[1-9][0-9][0-9][0-9][0-9][0-9][0-9][0-9][0-9]|-?1[0-9][0-9][0-9][0-9][0-9][0-9][0-9][0-9][0-9]|-?20[0-9][0-9][0-9][0-9][0-9][0-9][0-9][0-9]|-?21[0-3][0-9][0-9][0-9][0-9][0-9][0-9][0-9]|-?214[0-6][0-9][0-9][0-9][0-9][0-9][0-9]|-?2147[0-3][0-9][0-9][0-9][0-9][0-9]|-?21474[0-7][0-9][0-9][0-9][0-9]|-?214748[0-2][0-9][0-9][0-9]|-?2147483[0-5][0-9][0-9]|-?21474836[0-3][0-9]|[0-9]|214748364[0-7]") \
```

Comment:
What is happening here? Please add some comments.

Reply:
basically it only looks scary because the regex for longs is... long :)
i have a smaller explanation in the readme which uses the demonstration on a unsigned short, its the same principle here